prog/mutation.go

```diff
@@ -374,6 +374,29 @@ func (t *BufferType) mutate(r *randGen, s *state, arg Arg, ctx ArgCtx) (calls []
 	case BufferText:
 		data := append([]byte{}, a.Data()...)
 		a.data = r.mutateText(t.Text, data)
+	case BufferCompressed:
+		data := a.Data()
+		if len(data) == 0 {
+			return
+		}
+		data, err := Decompress(data)
+		if err != nil {
+			panic(fmt.Sprintf("could not decompress data: %v", err))
+		}
+		if len(data) == 0 {
+			return // Do not mutate empty data.
+		}
+		hm := MakeGenericHeatmap(data)
+		// At least one mutation, up to about one mutation every 256 KB
+		numMutations := r.Intn(len(data)/(1<<18)+1) + 1
+		for i := 0; i < numMutations; i++ {
+			index := hm.ChooseLocation(r.Rand)
+			width := 1 << uint(r.Intn(4))
+			if index+width <= len(data) {
```

Comment:
Probably not very important since we won't hit end of the image too often, but I would do the following instead just to not waste an iteration (in particular if we choose to do just 1 iteration):
```
if index+width > len(data) {
	width = 1
}
```